app/core/speaker_diarization.py

```
@@ -0,0 +1,312 @@
+            return filtered_intervals
+            
+        except Exception as e:
+            print(f"❌ fallback 화자분리 실패: {e}")
+            return []
+    
+    def _extract_speaker_number(self, speaker_id: str) -> int:
+        """화자 ID에서 숫자 추출"""
+        try:
+            if speaker_id.startswith("SPEAKER_"):
+                return int(speaker_id.split("_")[1])
+            return 0
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)